addons/mail/models/ir_actions.py

```diff
@@ -35,6 +42,14 @@ def _check_mail_thread(self):
             if action.state == 'followers' and not action.model_id.is_mail_thread:
                 raise ValidationError(_("Add Followers can only be done on a mail thread model"))
 
+    @api.constrains('state', 'model_id')
+    def _check_activity_mixin(self):
+        for action in self:
+            ActivityMixin = self.env['mail.activity.mixin']
+            ActionModel = self.env[action.model_id.model]
+            if action.state == 'next_activity' and ActivityMixin.__class__ not in ActionModel.__class__.__bases__:
```

Comment:
I saw in the code "issubclass(type(model), self.pool['mail.thread'])". Couldn't you use something similar ? Maybe just ask @rco-odoo  to be sure :) .

Reply:
🧐
I will ask him